### PR TITLE
Store the current filter's enrolled issuer list in cert-revocations

### LINKED ISSUE #241: Store the current filter's enrolled issuer list in cert-revocations
We're going to move the enrollment list from the `intermediates` collection to the `cert-revocations` collection.

The schema for `cert-revocations` has been updated with an `enrolledIssuers` field that holds a JSON array of strings. Each string in `enrolledIssuers` is to be a base64 encoded identifier for a CRLite-enrolled issuer. The identifier will be `SHA256(subject || spki)` where `subject` is a DER encoded RFC 5280 `RDNSequence` and `spki` is a DER encoded RFC 5280 `SubjectPublicKeyInfo`.

The CRLite aggregator should output the new identifiers in `enrolled.json`, and `moz_kinto_publisher` should populate the `enrolledIssuers` field. The signoff script should validate the `enrolledIssuers` field.

We'll continue to keep the `crlite_enrolled` flags in `intermediates` up to date for now.

Client-side changes will be tracked in [Bug 1750787](https://bugzilla.mozilla.org/show_bug.cgi?id=1750787)

### PR DESCRIPTION
**Pull request by jschanck**
Resolves #241 

~Note: The `crlite_enrollment_id.py` script is not used here. I needed it while making tests for the client-side code, and I think it makes more sense to keep it here with CRLite than mozilla-central.~ [edit: moved script to mozilla-central]

### CODE REVIEW

File: util/crlite_enrollment_id.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+if len(sys.argv) != 2:
+    print(f"Usage: {sys.argv[0]} <path to pem cert>")
+    sys.exit(1)
+
+with open(sys.argv[1], "r") as f:
+    cert = x509.load_pem_x509_certificate(f.read().encode("utf-8"))
+
+subj = cert.subject.public_bytes()
+spki = cert.public_key().public_bytes(
+    format=serialization.PublicFormat.SubjectPublicKeyInfo,
```
Comment (mozkeeler) [inline util/crlite_enrollment_id.py:26]: I'm a little concerned that we can't directly use the bytes of the certificate as-is, but I think if there's an encoding mismatch (which there shouldn't be, but, well...), we'll fall back to OCSP fetching, so this should be fine.

File: util/crlite_enrollment_id.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/python
```
Comment (mozkeeler) [inline util/crlite_enrollment_id.py:1]: Like I said in the other review, this should live where it's actually used. There's precedent for having utility scripts alongside tests in mozilla-central.

**Review by mozkeeler — APPROVED** [review]
Looks good. Just a couple of minor comments.

File: moz_kinto_publisher/main.py [hunk 3]
```diff
@@ -799,11 +799,18 @@ def publish_crlite_main_filter(
             }
         ]
 
+    enrolledIssuers = []
+    for issuer in intermediates:
+        if issuer["enrolled"]:
+            uid = base64.urlsafe_b64decode(issuer["uniqueID"])
+            enrolledIssuers.append(base64.b64encode(uid))
```
[the remark this answers is not in the document]
Comment (jschanck) [reply moz_kinto_publisher/main.py:806]: The aggregator-produced `enrolled.json` file uses urlsafe base64 for other values, so I did this for consistency. I'll open a separate bug to switch everything to standard base64; it's been on my TODO list.

File: util/crlite_enrollment_id.py (hunk 1 above)
Comment (jschanck) [reply util/crlite_enrollment_id.py:26]: This script is only used to generate test cases. The client and the aggregator get the bytes directly with moz::pkix and Go x509 respectively.

File: util/crlite_enrollment_id.py (hunk 2 above)
Comment (jschanck) [reply util/crlite_enrollment_id.py:1]: OK, I'll put a copy there.

**Review by mozkeeler — APPROVED** [review]
Sounds good!

**Pull request merged**